src/test/java/emissary/test/core/junit5/ExtractionTest.java

```diff
@@ -368,19 +368,53 @@ protected void checkAnswers(Document answers, IBaseDataObject payload, List<IBas
     protected void checkAnswers(Element el, IBaseDataObject payload, @Nullable List<IBaseDataObject> attachments, String tname)
             throws DataConversionException {
 
-        int numAtt = JDOMUtil.getChildIntValue(el, "numAttachments");
-        long numAttElements = el.getChildren().stream().filter(c -> c.getName().startsWith(ATTACHMENT_ELEMENT_PREFIX)).count();
+        int numAtt = -1;
+        long numAttElements = 0;
+        boolean osSpecificNumAtt = false;
+        List<Element> numAttachments = el.getChildren("numAttachments");
+        for (Element numAttEl : numAttachments) {
+            if (verifyOs(numAttEl)) {
+                numAtt = Integer.parseInt(numAttEl.getValue());
+                numAttElements = el.getChildren().stream().filter(
+                        c -> c.getName().startsWith(ATTACHMENT_ELEMENT_PREFIX) && verifyOs(c)).count();
+                // see if os specific numAttachments for check against payload
+                if (numAttEl.getAttribute("os-release") != null) {
+                    osSpecificNumAtt = true;
+                }
+                break;
+            }
+        }
+
         // check attachments answer file count against payload count
-        if (numAtt > -1) {
-            assertEquals(numAtt, attachments != null ? attachments.size() : 0,
-                    String.format("Expected <numAttachments> in %s not equal to number of att in payload.", tname));
-        } else if (numAtt == -1 && numAttElements > 0) {
-            assertEquals(numAttElements, attachments != null ? attachments.size() : 0,
-                    String.format("Expected <att#> in %s not equal to number of att in payload.", tname));
+        if (!osSpecificNumAtt) {
+            if (numAtt > -1) {
+                assertEquals(numAtt, attachments != null ? attachments.size() : 0,
+                        String.format("Expected <numAttachments> in %s not equal to number of att in payload.", tname));
+            } else if (numAtt == -1 && numAttElements > 0) {
+                assertEquals(numAttElements, attachments != null ? attachments.size() : 0,
+                        String.format("Expected <att#> in %s not equal to number of att in payload.", tname));
+            } else {
+                if (attachments != null && !attachments.isEmpty()) {
+                    fail(String.format("%d attachments in payload with no count in answer xml, add matching <numAttachments> count for %s",
+                            attachments.size(), tname));
+                }
+            }
         } else {
-            if (attachments != null && !attachments.isEmpty()) {
-                fail(String.format("%d attachments in payload with no count in answer xml, add matching <numAttachments> count for %s",
-                        attachments.size(), tname));
+            int attInPayload = attachments != null ? attachments.size() : 0;
+            assertTrue(numAtt <= attInPayload, String.format(
```

Comment:
Should this check be equals?

Reply:
removed and now just compares att# and numAttachments counts with the old comparisons